### PR TITLE
Add additional `expand_uid_template` unit tests

### PR DESCRIPTION
**Pull request by portante**
During the review of PR #3032 a few additional unit tests were identified.

### CODE REVIEW

**Review by ndokos — APPROVED** [review]

File: lib/pbench/test/unit/server/test_indexer.py [hunk 1]
```diff
@@ -22,12 +22,32 @@ def test_not_found():
...
+        res = ResultData.expand_uid_template(templ, dict(foo=1, bar=2))
+        assert res == templ
+
+        keyvals = {"benchmark_name": "mybm"}
         templ = "%benchmark_name%_UID"
         res = ResultData.expand_uid_template(templ, keyvals)
         assert res == "mybm_UID"
+
+        keyvals = {"name": "myname"}
+
```
Comment (dbutenhof) [inline lib/pbench/test/unit/server/test_indexer.py:35]: I found this blank line vaguely unsettling, until I realized you were using the same `keyvals` in all of the following tests. Perhaps a comment?

**Review by dbutenhof — DISMISSED** [review]

**Review by npalaska — DISMISSED** [review]

**Review by ndokos — APPROVED** [review]

File: lib/pbench/test/unit/server/test_indexer.py [hunk 2]
```diff
@@ -22,12 +22,33 @@ def test_not_found():
...
     def test_fallbacks():
-        keyvals = {"name": "mybm"}
+        templ = "%benchmark_name%_UID"
+        res = ResultData.expand_uid_template(templ, dict(foo=1, bar=2))
+        assert res == templ
+
+        keyvals = {"benchmark_name": "mybm"}
         templ = "%benchmark_name%_UID"
         res = ResultData.expand_uid_template(templ, keyvals)
         assert res == "mybm_UID"
```
Comment (webbnh) [inline lib/pbench/test/unit/server/test_indexer.py:32]: Not to quibble, but why the difference in idiom between these two test cases?  (I.e., the first specifies the dictionary in-line, while the second sets up a single-use local variable....)  Ditto for lines 46 and 51.

And, as long as I'm at it, why reassign `templ` every time if the value is the same in each case?  (Perhaps use a single assignment?  Perhaps just in-line the value?)

Comment (webbnh) [inline lib/pbench/test/unit/server/test_indexer.py:32]: I think this case is redundant, given `test_found()`, but I s'pose it doesn't hurt.

**Review by webbnh — APPROVED** [review]
These changes are OK, but they are uneven:  in some places you use in-line values for arguments, in some places you use single-use local variables, in some places you reuse local variables, and in some places you redefine the same local variable with the same value.  (And, one of the test cases is redundant.)

So, I'm OK with the code as is, but it would be better if it were polished a bit.

**Review by dbutenhof — DISMISSED** [review]

**Review by webbnh — APPROVED** [review]
👍

**Review by dbutenhof — APPROVED** [review]

**Pull request merged**